airflow/bin/cli.py

```diff
@@ -414,7 +414,7 @@ def variables_export(args):
     _vars_wrapper(args, export=args.file)
 
 
-@cli_utils.deprecated_action(new_name='variables')
+@cli_utils.deprecated_action(new_name='variables list')
```

Comment:
This does not make sense to me:

```
root@99624d92186f:/opt/airflow# airflow variables
The 'variables' command is deprecated and removed in Airflow 2.0, please use 'variables list' instead

root@99624d92186f:/opt/airflow# airflow variables list
usage: airflow variables [-h] COMMAND ...

Manage variables

positional arguments:
  COMMAND
    delete    Delete variable
    export    Export all variables
    get       Get variable
    import    Import variables
    set       Set variable

optional arguments:
  -h, --help  show this help message and exit

airflow variables command error: argument COMMAND: invalid choice: 'list' (choose from 'delete', 'export', 'get', 'import', 'set'), see help above.
```

Reply:
Hi @kaxil , that's because `variables list` is missed in the new Variable Commands in 1.10.14 (https://github.com/apache/airflow/blob/1.10.14/airflow/bin/cli.py#L3066), but it is a valid new command in 2.0 (https://github.com/apache/airflow/blob/2.0.1/airflow/cli/cli_parser.py#L1037)

Reply:
In such a case, we will also need to make further change to ensure `variables list` present in 1.10.x (say 1.10.15) as well

Reply:
Old style:

`airflow variables -l`

New

`airflow variables list` etc.

Reply:
I will shortly prepare a PR against branch v1-10-stable to add `airflow variables list`.

Meanwhile, I think this PR is ok to be merged.

Reply:
Prepared PR https://github.com/apache/airflow/pull/14462 to supplement this, as mentioned above.

Reply:
Cool, approved  #14462